roseau/load_flow/models/grounds.py

```diff
@@ -127,3 +129,248 @@ def _to_dict(self, include_results: bool) -> JsonDict:
+        connected_phases = [gc._phase for gc in ground._connections if gc._element is element and gc._side == side]
+        if connected_phases:
+            pretty_phase = pretty_phases.replace("phases", "phase")
+            if phase in connected_phases:
+                msg = (
+                    f"Ground {ground.id!r} is already connected to {pretty_phase} {phase!r} of {element._element_info}."
+                )
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_PHASE)
+            if on_connected != "ignore":
+                connections, _ = one_or_more_repr(connected_phases, pretty_phase, pretty_phases)
+                msg = f"Ground {ground.id!r} is already connected to {connections} of {element._element_info}."
+                if on_connected == "raise":
+                    logger.error(msg)
+                    raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_GROUND_ID)
+                else:
+                    warnings.warn(msg, stacklevel=find_stack_level())
+
+        super().__init__(id)
+        self._connect(ground, element)
+        self._ground = ground
+        self._element = element
+        self._phase = phase
+        self._side: Side = side
+        self.on_connected: Literal["raise", "warn", "ignore"] = on_connected
+        self.impedance = impedance
+        ground._connections.append(self)
+
+        self._res_current: complex | None = None
+
+    def __repr__(self) -> str:
+        parts = [
+            f"id={self.id!r}",
+            f"ground={self._ground.id!r}",
+            f"element={self._element.id!r}",
+            f"impedance={self._impedance!r}",
+            f"phase={self._phase!r}",
+        ]
+        if self._side is not None:
+            parts.append(f"side={self._side!r}")
+        parts.append(f"on_connected={self.on_connected!r}")
+        return f"<{type(self).__name__}: {', '.join(parts)}>"
+
+    @property
+    def phase(self) -> str:
+        """The phase of the connection to the ground."""
+        return self._phase
+
+    @property
+    def side(self) -> Side:
+        """The side of the element to connect to."""
+        return self._side
+
+    @property
+    @ureg_wraps("ohm", (None,))
+    def impedance(self) -> Q_[complex]:
+        """The impedance of the connection to the ground (ohm)."""
+        return self._impedance  # type: ignore
+
+    @impedance.setter
+    @ureg_wraps(None, (None, "ohm"))
+    def impedance(self, value: Complex | Q_[Complex]) -> None:
+        self._impedance = complex(value)
+        self._invalidate_network_results()
+        if np.isclose(self._impedance, 0):
+            if not isinstance(self._cy_element, CySwitch):
+                if self._cy_element is not None:
+                    self._cy_element.disconnect()
+                self._cy_element = CySwitch(n=1)
+                self._cy_connect()
```

Comment:
As we change the `CyElement` here, the `CyElectricalNetwork` of the network isn't valid anymore. We have to mark the network as invalid (as we do when we add a short circuit for example) with 
```python
if self.network is not None:
    network._valid = False
```